date: build failure of cygwin target

```
- name: "`make install MULTICALL=y` (x86_64-pc-cygwin)"
  shell: 'C:\msys64\usr\bin\bash.exe --login -eo pipefail {0}'
  env:
    CHERE_INVOKING: 1
  run: |
    set -x
    pacman -Sy --noconfirm --needed make rust base-devel
    cargo check -p uu_date

    Checking uu_date v0.8.0 (/d/a/coreutils/coreutils/src/uu/date)
error[E0425]: cannot find value `D_T_FMT` in crate `libc`
  --> src/uu/date/src/locale.rs:33:43
   |
33 |     const DATE_FMT: libc::nl_item = libc::D_T_FMT;
   |                                           ^^^^^^^ not found in `libc`
error[E0425]: cannot find function `nl_langinfo` in crate `libc`
  --> src/uu/date/src/locale.rs:78:37
   |
78 |             let d_t_fmt_ptr = libc::nl_langinfo(DATE_FMT);
   |                                     ^^^^^^^^^^^ not found in `libc`
```
libc v0.2.186
@tgross35

--- a/.github/workflows/make.yml
+++ b/.github/workflows/make.yml
@@ -303,7 +303,7 @@ jobs:
       run: |
         set -x
         pacman -Sy --noconfirm --needed make rust base-devel
-        DESTDIR=/tmp/c make install PROG_PREFIX=uu- PROFILE=release-small COMPLETIONS=n MANPAGES=n LOCALES=n SKIP_UTILS=date # FIXME: build date
+        DESTDIR=/tmp/c make install PROG_PREFIX=uu- PROFILE=release-small COMPLETIONS=n MANPAGES=n LOCALES=n
         # Check that utils are built with given profile
         ./target/release-small/true
         # Check that the progs have prefix

--- a/src/uu/date/src/locale.rs
+++ b/src/uu/date/src/locale.rs
@@ -12,7 +12,7 @@
 macro_rules! cfg_langinfo {
     ($($item:item)*) => {
         $(
-            #[cfg(all(unix, not(target_os = "android"), not(target_os = "redox")))]
+            #[cfg(all(unix, not(target_os = "android"), not(target_os = "cygwin"), not(target_os = "redox")))]
             $item
         )*
     }
@@ -27,9 +27,9 @@ cfg_langinfo! {
 
     /// glibc's `_DATE_FMT` has been stable for the last 12 years
     /// being added upstream to libc TODO: update to libc
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "cygwin"))]
     const DATE_FMT: libc::nl_item = 0x2006c;
-    #[cfg(not(target_os = "linux"))]
+    #[cfg(not(any(target_os = "linux", target_os = "cygwin")))]
     const DATE_FMT: libc::nl_item = libc::D_T_FMT;
 }
 
@@ -110,7 +110,12 @@ cfg_langinfo! {
 }
 
 /// On platforms without nl_langinfo support, use 24-hour format by default
-#[cfg(any(not(unix), target_os = "android", target_os = "redox"))]
+#[cfg(any(
+    not(unix),
+    target_os = "android",
+    target_os = "cygwin",
+    target_os = "redox"
+))]
 pub fn get_locale_default_format() -> &'static str {
     "%a %b %e %X %Z %Y"
 }

